### ISSUE TITLE
first pass refactor `garden-ai prototype` commands/interface

### ISSUE THREAD

**Issue by OwenPriceSkelly** [Garden-AI/garden#306]
Scope of this ticket is a little loose but primarily focused on early quality of life improvements to the interface we want to prioritize for the sake of the development process 

Broad strokes:
  - cli commands should indicate the 1:1 relationship between notebooks and container images
    - if I point the cli at a notebook to build an image, pointing it at the same notebook later should reuse the image etc
    - maybe just use the notebook name for the container name here 
  - better support to "see what got planted" for debugging purposes while still in-notebook
    - (maybe in the same notebook with the session loaded instead of running the cells? see what works) 
  - part of setting up garden should be setting up/ verifying where the containers should be pushed 
  - anything else that comes up I'll add here

**Comment by isaac-darling** [comment]
There are a couple of things here that I am unsure about. For instance, an image being "reused" later can mean two things:

a. The notebook hasn't changed, in which case, why would the CLI be pointed at it again?
b. The notebook has changed, in which case, the container must NOT be reused because it will not reflect those changes.

Of course if I misinterpreted you somewhere, please let me know!

**Comment by OwenPriceSkelly** [comment]
Ah yeah sorry this issue is really just what I had jotted down after a huddle the other day, so its pretty provisional and missing some context. 

The bit about reuse was thinking more about editing/developing the notebook before finalizing it with the plant/finalize/publish step (e.g. if I copy a `my_model_definition.py` file over when I make the notebook, maybe I shouldn't need to explicitly copy it every time I want to spin up that notebook's container)

the answer to a) though is debugging -- a notebook that worked fine before planting/publishing doesn't work when I try running it on an endpoint. we decided that dropping someone who thought they were debugging a notebook into a vanilla python repl is probably not the way to go, but we also don't want to put them in a context that diverges too far from what the endpoint actually sees. the dream would be like the same notebook but it spits you out in a new cell at the bottom, and all the others are grayed out in favor of a `load_session`, maybe? very much still noodling with it
